ui/docs/manual/deploying/database.mdx

```diff
@@ -0,0 +1,89 @@
+---
+title: Database Configuration
+---
+
+import Warning from 'taskcluster-ui/views/Documentation/components/Warning';
+
+# Database Configuration
+
+Taskcluster uses a Postgres 11 database for its backend storage.
+A single database is shared among all services.
+
+Taskcluster assumes that it "owns" the database, but can share a single server with other users.
+For production purposes we recommend a dedicated server, due to the possibility of contention for server resources, but sharing a server acceptable for non-production environments.
+
+## DB Users
+
+Each service uses its own Postgres user to access the database, and Taskcluster uses this functionality internally to isolate services from one another.
+The Taskcluster deployment process additionally requires an "admin" user that is used for schema migrations and to update the permissions of the per-service users.
+This admin user must have full permission to all resources in the selected database, as well as permission to grant and revoke access for the service users.
+
+The admin user can have any name.
+Because users are global to a Postgres server, Taskcluster requires a name prefix which is applied to each per-service user.
+In a non-production environment, this allows several installations of Taskcluster to co-exist on the same server.
+We recommend that the admin username and the prefix be the same string.
+
+The set of users that must be configured, then, are:
+
+<!-- USERLIST BEGIN -->
+ * `<prefix>` -- admin user
+ * `<prefix>_secrets` -- user for Taskcluster secrets service
+<!-- USERLIST END -->
+
+Set these up using something with the following effect:
+
+```
+CREATE USER <prefix> PASSWORD '<admin_password>';
+GRANT ALL ON DATABASE <dbname> TO <prefix> WITH GRANT OPTION;
+CREATE USER <prefix>_taskcluster_queue PASSWORD '<queue_password>';
```

Comment:
Wasn't the consensus to have the password for each service created internally using a randomized algorithm?

Reply:
(we also just talked about this in zoom) my take-away from that conversation was that they should be generated randomly, but we didn't discuss in detail what should do the generation.  If Taskcluster does that generation internally, then it needs some way to feed those passwords "upstream" the the deployment infrastructure where they can be deployed to the services.  The approach implemented here is similar to that for pulse and Azure: those doing the deployment must pre-create the accounts and supply the credentials, but Taskcluster manages everything after that.